lib/src/arithmetic/signals/fixed_point_logic.dart

```diff
@@ -123,6 +123,13 @@ class FixedPoint extends Logic {
     }
   }
 
+  /// Multiply
+  Logic fpMultiply(dynamic other) {
+    _verifyCompatible(other);
+    final product = Multiply(this, other).out;
+    return FixedPoint.of(product, signed: false, m: product.width - n, n: n);
```

Comment:
since `Multiply` has the same output width as inputs, `product.width - n` should be the same as `m` right?  Or is it 1 bigger due to sign bit (if signed)?

Something to consider: do we want `FixedPoint` to widen or keep the same width when multiplied by another `FixedPoint`?  To keep with API consistency, it seems reasonable to keep the same width, since `FixedPoint` is a type of `Logic`.  But do we need a way to appropriately "expand" a `FixedPoint`, sort of like a `signExtend` but that makes it the "right" shape (m and n) for an output of a multiplication?

This is probably beyond the scope of this PR, but since this change is here and you're the first user of this new `*`, thought I'd ask for thoughts :)

Reply:
So, I spent a lot of time on this last night. You do want your Fixed point to grow, or you loose A LOT of precision. Specifically, when we multiply to `N` values we end up with `2*N - 1` result. Because our n doesn't move, that means our m grows by the difference of the output width and n: `m = total_width - n`. So you could sign extend, but you will lose data. If you look at most multipliers in general your product usually grows with it. You can than leave it to the user to truncate how they will.

I'm hesitant to force it to the same size as the incoming products because you just loose so much information. Instead of opting to give precise precision to the user to then truncate as they want (or make an override to do this?)

Reply:
I did my math here wrong (darn being out of practice for so long). It is still 2*N-1 but the results are `2*m` and `2*n`. So yes, we either need to sign extend or not. Going to push a fix for that.

Reply:
So for example, if we do 3.5 * 1.25 we expect 4.375. These numbers are (11.10 *01.01) = 0100.0110. If we truncate that from the Q4.4 format back to Q2.2 format we get 00.01 which is 0.5 and inaccurate. So I DO need to fix this to be `m: 2*m, n: 2*n` but I think we should keep it widened. (Never not double your check after coding late at night so I'm glad you called this out).

Reply:
Size was fixed to 2m.2n